mlir/test/Ion/QuantumToIon.mlir

```diff
@@ -0,0 +1,134 @@
+                >,
+                einstein_a=10.10
+            >,
+            #ion.transition<
+                level_0 = #ion.level<
+                    principal=1,
+                    spin=1.1,
+                    orbital=2.2,
+                    nuclear=3.3,
+                    spin_orbital=4.4,
+                    spin_orbital_nuclear=5.5,
+                    spin_orbital_nuclear_magnetization=6.6,
+                    energy=8.8
+                >,
+                level_1 = #ion.level<
+                    principal=1,
+                    spin=1.1,
+                    orbital=2.2,
+                    nuclear=3.3,
+                    spin_orbital=4.4,
+                    spin_orbital_nuclear=5.5,
+                    spin_orbital_nuclear_magnetization=6.6,
+                    energy=8.8
+                >,
+                einstein_a=10.10
+            >
+        ]
+    }: !ion.ion
+
+    %1 = quantum.alloc( 2) : !quantum.reg
+    %2 = quantum.extract %1[ 0] : !quantum.reg -> !quantum.bit
+    %3 = quantum.extract %1[ 1] : !quantum.reg -> !quantum.bit
+
+    // CHECK: [[rx1out:%.+]] = ion.parallelprotocol(%2) : !quantum.bit {
+    // CHECK-NEXT: ^{{.*}}(%arg1: !quantum.bit):
+    // CHECK-NEXT: ion.pulse(%arg0 : f64) %arg1 {beam = #ion.beam<transition_index = 0 : i64, rabi = 1.100000e+00 : f64, detuning = 2.200000e+00 : f64, polarization = dense<[0, 1]> : vector<2xi64>, wavevector = dense<[-2, 3]> : vector<2xi64>>, phase = 0.00{{.*}} : f64}
+    // CHECK-NEXT: ion.pulse(%arg0 : f64) %arg1 {beam = #ion.beam<transition_index = 1 : i64, rabi = 1.100000e+00 : f64, detuning = 2.200000e+00 : f64, polarization = dense<[0, 1]> : vector<2xi64>, wavevector = dense<[-2, 3]> : vector<2xi64>>, phase = 0.00{{.*}} : f64}
+    // CHECK-NEXT: }
+    %4 = quantum.custom "RX"(%arg0) %2 : !quantum.bit
+
+    // CHECK: [[ry1out:%.+]] = ion.parallelprotocol([[rx1out]]) : !quantum.bit {
+    // CHECK-NEXT: ^{{.*}}(%arg1: !quantum.bit):
+    // CHECK-NEXT: ion.pulse(%arg0 : f64) %arg1 {beam = #ion.beam<transition_index = 0 : i64, rabi = 1.100000e+00 : f64, detuning = 2.200000e+00 : f64, polarization = dense<[0, 1]> : vector<2xi64>, wavevector = dense<[-2, 3]> : vector<2xi64>>, phase = 0.00{{.*}} : f64}
+    // CHECK-NEXT: ion.pulse(%arg0 : f64) %arg1 {beam = #ion.beam<transition_index = 1 : i64, rabi = 1.100000e+00 : f64, detuning = 2.200000e+00 : f64, polarization = dense<[0, 1]> : vector<2xi64>, wavevector = dense<[-2, 3]> : vector<2xi64>>, phase = 3.14{{.*}} : f64}
+    // CHECK-NEXT: }
+    %5 = quantum.custom "RY"(%arg0) %4 : !quantum.bit
+
+    // CHECK: [[rx2out:%.+]] = ion.parallelprotocol([[ry1out]]) : !quantum.bit {
+    // CHECK-NEXT: ^{{.*}}(%arg1: !quantum.bit):
+    // CHECK-NEXT: ion.pulse(%arg0 : f64) %arg1 {beam = #ion.beam<transition_index = 0 : i64, rabi = 1.100000e+00 : f64, detuning = 2.200000e+00 : f64, polarization = dense<[0, 1]> : vector<2xi64>, wavevector = dense<[-2, 3]> : vector<2xi64>>, phase = 0.00{{.*}} : f64}
```

Comment:
Convert these long `CHECK` lines to `CHECK-SAME`

Reply:
0c316e2

Reply:
Thanks!